packages/flutter/test/material/floating_action_button_location_test.dart

```diff
@@ -296,6 +296,21 @@ void main() {
     expect(tester.getCenter(find.byType(FloatingActionButton)), const Offset(756.0, 572.0));
   });
 
+  testWidgets('Contained floating action button locations', (WidgetTester tester) async {
+    await tester.pumpWidget(
+      _buildFrame(
+        location: FloatingActionButtonLocation.endContained,
+        bab: const SizedBox(height: 100.0),
+        viewInsets: EdgeInsets.zero,
+      ),
+    );
+
+    // Scaffold 800x600, FAB is 56x56, BAB is 800x100, FAB's center is
+    // at the top of the BAB.
+    // Formula: scaffold height - BAB height + FAB height / 2 + BAB top & bottom margins.
```

Comment:
I'm not following this but it's probably my fault.  `600 - 100 + 56/2 + 100 + 100` should equal  what?  Or what are the margins?

Reply:
Well I dropped this comment just to make easy understand what is the expected.

The values are: `600 - 100 + 28 + 22`